### DISCUSSION TITLE
Clarifying documentation - On async calls

### DISCUSSION THREAD

**Discussion by nemosmithasf** [101arrowz/fflate#176] [General]
`if you call unzip ten times, the overhead only applies for the first call, but if you call unzip and zlib, they will each cause the 70ms delay.`

Is this implying that we are reusing the workers if we call the same function consecutively, but we are re-instantiating workers if we call alternating functions? i.e. the function keeps track of worker types in the background?

**Comment by 101arrowz** [comment]
Actually the worker code is generated the first time, and after that the code is cached but not the worker instance. This could be probably be changed to actually cache the worker instance to improve performance.

**Reply by nemosmithasf** [reply]
Ah, so basically if we called unzip and zlib each 10 times, the expected overhead is 2 * 70ms, not 2 * 70 * 10ms

**Reply by 101arrowz** [reply]
Yeah that's correct
